internal/datastore/postgres/revisions.go

```diff
@@ -0,0 +1,125 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/shopspring/decimal"
+
+	"github.com/authzed/spicedb/internal/datastore"
+)
+
+const (
+	errRevision      = "unable to find revision: %w"
+	errCheckRevision = "unable to check revision: %w"
+
+	// querySelectRevision will round the database's timestamp down to the nearest
+	// quantization period, and then find the first transaction after that. If there
+	// are no transactions newer than the quantization period, it just picks the latest
+	// transaction
+	querySelectRevision = `
+	SELECT COALESCE(
```

Comment:
I wonder whether we should have a test that explicitly runs these queries and validates their outputs?

Reply:
Added.